### ISSUE TITLE
ament_clang_tidy doesn't show errors, thinks all headers are system headers

### ISSUE THREAD

**Issue by rotu** [ament/ament_lint#161]
I'm trying to use the new `ament_clang_tidy`, but it's not (1) explaining the errors generated (2) emitting the 96357 warnings it thinks are in non-user code. I'm running Python 3.6.8 

```ament_clang_tidy src/ros2/demos```

> 1 error generated.
> Error while processing /opt/ros/master/src/ros2/demos/composition/include/composition/client_component.hpp.
> 2 errors generated.
> Error while processing /opt/ros/master/src/ros2/demos/composition/include/composition/listener_component.hpp.
> 3 errors generated.
> Error while processing /opt/ros/master/src/ros2/demos/composition/include/composition
> ...
> Error while processing /opt/ros/master/src/ros2/demos/quality_of_service_demo/rclcpp/src/liveliness.cpp.
> 95568 warnings and 71 errors generated.
> Error while processing /opt/ros/master/src/ros2/demos/quality_of_service_demo/rclcpp/src/utils.cpp.
> 96367 warnings and 72 errors generated.
> Error while processing /opt/ros/master/src/ros2/demos/quality_of_service_demo/rclcpp/src/utils.hpp.
> Suppressed 96357 warnings (96357 in non-user code).
> Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.

**Comment by rotu** [comment]
@JShep1

**Comment by JShep1** [comment]
In the meantime while I resolve this issue, you can use `--xunit-file out.xml` to get instant feedback on the generated errors in xunit convention.  

I'll look into adding the error/warning output.

Not sure what you mean by it thinks all headers are system headers given , could you elaborate?

**Comment by rotu** [comment]
Here’s what I mean:

> 96367 warnings and 72 errors generated.
...
> Suppressed 96357 warnings (96357 in non-user code).

**Comment by JShep1** [comment]
There's a bug I'm currently tracking down that will definitely resolve the first issue and may be causing the second as well - I'll keep you posted.

**Comment by JShep1** [comment]
It seems as if this functionality is purposeful and a side effect of the way `clang-tidy` works, see [here](https://stackoverflow.com/questions/46638293/ignore-system-headers-in-clang-tidy).  `clang-tidy` delves into all included headers and checks those recursively.

The errors you are likely getting are typically system header files that are not found, be sure to have clang installed on the testing system, see second most recent comment [here](https://stackoverflow.com/questions/20587228/clang-error-stddef-file-not-found)

Additionally, `clang-tidy` operates within the scope it's given.  `*.h not found` errors are likely because the header file(s) resides somewhere outside of the scope that it is operating within.

I'm going to make a PR putting in stdout and adding non-system header output option, but it doesn't seem as if there's a way to quash warning output from system headers completely.

**Comment by JShep1** [comment]
Issue resolved in https://github.com/ament/ament_lint/pull/163 .  Going to close the issue for now, feel free to reopen if needed.

**Comment by rotu** [comment]
@JShep1 yes, I know it's supposed to skip warnings in system headers. I just find it implausible that *none* of the errors are in my code :-).

Reopening since the fix PR #163 is not yet merged. Add "`fixes #161`" to the PR description to [close this issue automatically when the PR is merged](https://help.github.com/en/articles/closing-issues-using-keywords).

**Comment by rotu** [comment]
Oops. I can't reopen the issue for some reason?
